presto-orc/src/main/java/com/facebook/presto/orc/writer/TimestampColumnWriter.java

```diff
@@ -121,7 +121,11 @@ public void writeBlock(Block block)
 
                 long seconds = (value / MILLIS_PER_SECOND) - baseTimestampInSeconds;
                 long millis = value % MILLIS_PER_SECOND;
-                long encodedNanos = millis == 0 ? 0 : millis << 3 | MILLIS_TO_NANOS_TRAILING_ZEROS;
+                // Because the number of nanoseconds often has a large number of trailing zeros, the trailing
+                // decimal zero digits are removed, and the last three bits are used to record how many zeros
+                // were removed (minus one). Thus 1,000,000 nanoseconds would be serialized as '1' with `5` in
+                // the last three bytes: 0b0000_1110.
```

Comment:
While this comment is kind of good enough for our case (because we only care about 6 trailing zeros). When I read this comment, the first thing that come into mind is how a nano would be serialized if there are no trailing decimal zero digits. And I would be suspicious about the correctness/accuracy of this comment. As a result, I would recommend making this comment more accurate.